go-controller/pkg/ovn/controller/services/node_tracker.go

```diff
@@ -147,28 +161,22 @@ func (nt *nodeTracker) updateNodeInfo(nodeName, switchName, routerName string, n
 	}
 
 	nt.nodes[nodeName] = ni
-	nt.Unlock()
 
-	klog.Infof("Node %s switch + router changed, syncing services", nodeName)
 	// Resync all services
-	nt.resyncFn()
-}
-
-// removeNodeWithServiceReSync removes a node from the LB -> node mapper
-// *and* forces full reconciliation of services.
-func (nt *nodeTracker) removeNodeWithServiceReSync(nodeName string) {
-	nt.removeNode(nodeName)
-	nt.resyncFn()
+	nt.resyncFn(nt.allNodes())
 }
 
 // RemoveNode removes a node from the LB -> node mapper
-// We don't need to re-sync here, because any stale LBs
-// will eventually be cleaned up, and they don't have any cost.
-func (nt *nodeTracker) removeNode(nodeName string) {
+// If 'doResync' is true then this also forces a full reconciliation of
+// services.
+func (nt *nodeTracker) removeNode(nodeName string, doResync bool) {
```

Comment:
nit: I don't know if we added this boolean here and removed `removeNodeWithServiceReSync` for a reason that is no longer here because now it looks like a superfluous change. In this specific case, I did prefer the explicit naming `removeNodeWithServiceReSync` rather than a boolean that you have to chase down to see what it means. Up to you.

Reply:
I changed it back, it's more clear indeed.